python/adbc_driver_manager/adbc_driver_manager/dbapi.py

```diff
@@ -40,25 +49,22 @@
 
 try:
     import pyarrow
-except ImportError as e:
-    raise ImportError("PyArrow is required for the DBAPI-compatible interface") from e
-
-try:
     import pyarrow.dataset
 except ImportError:
-    _pya_dataset = ()
-    _pya_scanner = ()
+    _has_pyarrow = False
 else:
-    _pya_dataset = (pyarrow.dataset.Dataset,)
-    _pya_scanner = (pyarrow.dataset.Scanner,)
+    _has_pyarrow = True
+    from . import _reader
 
 import adbc_driver_manager
 
-from . import _lib, _reader
+from . import _lib
 from ._lib import _blocking_call
 
 if typing.TYPE_CHECKING:
     import pandas
+    import polars
+    import pyarrow
```

Comment:
don't we need another try/except here too?

Reply:
This is under TYPE_CHECKING so it's only run under pyright, and pyright would complain anyways if you're missing these